cpp/src/io/orc/stripe_data.cu

```diff
@@ -132,6 +132,146 @@ struct orcdec_state_s {
   } vals;
 };
 
+/**
+ * @brief Manage caching of the first run of TIMESTAMP's DATA stream for a row group.
+ *
+ * This class is used to address a special case, where the first run spans two adjacent row groups
+ * and its length is greater than the maximum length allowed to be consumed. This limit is imposed
+ * by the decoder when processing the SECONDARY stream. This class shall be instantiated in the
+ * shared memory. As an optimization, the actual cache is a local variable and does not reside in
+ * the shared memory.
+ */
+class run_cache_manager {
+ private:
+  enum class status : uint8_t {
+    DISABLED,             ///< Run cache manager is disabled. No caching will be performed.
+    CAN_WRITE_TO_CACHE,   ///< Run cache manager is ready for write. This is expected to happen in
+                          ///< the first iteration of the top-level while-loop in
+                          ///< gpuDecodeOrcColumnData().
+    CAN_READ_FROM_CACHE,  ///< Run cache manager is ready for read. This is expected to happen in
+                          ///< the second iteration of the while-loop.
+  };
+
+ public:
+  /**
+   * @brief Initialize the run cache manager.
+   *
+   * @param[in] s ORC decoder state.
```

Comment:
No need to mention `[in/out]` for `@param` if all of them have the same direction.